hasjob/views/listing.py

```diff
@@ -583,9 +583,17 @@ def rejectjob(domain, hashid):
                 post.domain.banned_reason = rejectform.reason.data
                 if post.user:
                     post.user.blocked = True
+
+                # In case of Ban, Reject all the job posts by the same domain
+                # unless it's a webmail domain
+                if not post.domain.is_webmail:
+                    for jobpost in post.domain.jobposts:
+                        if jobpost.is_listed():
+                            jobpost.status = POSTSTATUS.REJECTED
```

Comment:
You should also set these fields: `closed_datetime`, `review_comments`, `review_datetime` and `reviewer`. See above for how they're set on the current post and use the same values for all of these.